### PR TITLE
Add load balancing and IPC to gRPC performance

### PR DESCRIPTION
**Pull request by JamesNK**
Add more sections to gRPC perf page

### CODE REVIEW

File: aspnetcore/grpc/performance.md [hunk 1]
```diff
@@ -70,8 +70,40 @@ There are a couple of workarounds for .NET Core 3.1 apps:
...
+
+`Grpc.Net.Client` currently doesn't support client-side load balancing. The [Grpc.Core](https://www.nuget.org/packages/Grpc.Core) is a good choice if client-side load balancing is required in .NET.
+
+### Proxy load balancing
+
+An L7 (application) proxy works at a higher level than an L4 (transport) proxy. L7 proxies understand HTTP/2, and are able to distribute gRPC calls multiplexed to the proxy on one HTTP/2 connection across multiple endpoints. Using a proxy is simpler than client-side load balancing, but can add extra latency to gRPC calls.
+
+There are many L7 proxies available. Two options are:
+
+1. [Envoy](https://www.envoyproxy.io/) proxy - A popular open source proxy.
```
Comment (davidfowl) [inline aspnetcore/grpc/performance.md:96]: https://linkerd.io/

Comment (davidfowl) [reply aspnetcore/grpc/performance.md:96]: Maybe some mention of kubernetes

File: aspnetcore/grpc/performance.md [hunk 2]
```diff
@@ -70,8 +70,43 @@ There are a couple of workarounds for .NET Core 3.1 apps:
...
+
+Lookaside client load balancing is a technique where load balancing state is stored in a central location. Clients periodically query the central location for information to use when making load balancing decisions.
+
+`Grpc.Net.Client` currently doesn't support client-side load balancing. The [Grpc.Core](https://www.nuget.org/packages/Grpc.Core) is a good choice if client-side load balancing is required in .NET.
+
+### Proxy load balancing
+
+An L7 (application) proxy works at a higher level than an L4 (transport) proxy. L7 proxies understand HTTP/2, and are able to distribute gRPC calls multiplexed to the proxy on one HTTP/2 connection across multiple endpoints. Using a proxy is simpler than client-side load balancing, but can add extra latency to gRPC calls.
```
Comment (Tratcher) [inline aspnetcore/grpc/performance.md:94]: Note even an L7 proxy will usually only be able to load balance distinct requests. gRPC messages streamed on a single request would not be load balanced.

Comment (JamesNK) [reply aspnetcore/grpc/performance.md:94]: That's true. At least in this doc I don't want to go into that level of detail.

**Pull request merged**